storage/store.go

```diff
@@ -978,27 +1026,33 @@ func (s *Store) AddRange(rng *Range) error {
 	return nil
 }
 
-// addRangeInternal adds the range to the ranges map and the rangesByKey slice.
-// If resort is true, the rangesByKey slice is sorted; this is optional to
-// allow many ranges to be added and the sort only invoked once. This method
-// presupposes the store's lock is held. Returns a rangeAlreadyExists error if
-// a range with the same Raft ID has already been added to this store.
-func (s *Store) addRangeInternal(rng *Range, resort bool) error {
+// addRangeInternal adds the range to the ranges map and the rangesByKey btree.
+// Returns a rangeAlreadyExists error if a range with the same Raft ID
+// has already been added to this store.
+func (s *Store) addRangeInternal(rng *Range) error {
 	// TODO(spencer); will need to determine which range is
 	// newer, and keep that one.
 	if exRng, ok := s.ranges[rng.Desc().RaftID]; ok {
 		return &rangeAlreadyExists{exRng}
 	}
 	s.ranges[rng.Desc().RaftID] = rng
-	s.rangesByKey = append(s.rangesByKey, rng)
-	if resort {
-		sort.Sort(s.rangesByKey)
+	rngItem := (*rangeBtreeItem)(rng)
+	if s.rangesByKey.Has(rngItem) {
```

Comment:
Hmm, no, the test just explicitly adds a range that has the same range but has a different Raft ID (`rng2Dup`)..

All the other tests passed even if we remove this `Has` check. So should be fine with just tweaking `TestStoreAddRemoveRanges`?